src/certgen_certificate_generator_server.cc

```diff
@@ -27,11 +27,285 @@
+        std::stringstream configJson;
+        configJson << configFile.rdbuf();
+        configFile.close ();
+
+        cxxtools::SerializationInfo certgenSi;
+        cxxtools::JsonDeserializer deserializer (configJson);
+        deserializer.deserialize (certgenSi);
+
+        certgenSi >>= certgenConfig;
+
+        fty::CertificateConfig config;
+        // fill the configuration data
+        uint8_t version;
+        std::istringstream versionStr (certgenConfig.version());
+        versionStr >> version;
+        config.setVersion(version);
+
+        std::chrono::hours validityOffset (HOURS_IN_DAY * certgenConfig.certConf().getValidityOffset());
+        std::chrono::time_point<std::chrono::system_clock, std::chrono::seconds> currentTimestamp =
+            std::chrono::time_point_cast<std::chrono::seconds>(std::chrono::system_clock::now());
+        std::chrono::time_point<std::chrono::system_clock, std::chrono::seconds> startTimestamp = currentTimestamp + validityOffset;
+        config.setValidFrom(startTimestamp.time_since_epoch().count());
+
+        std::chrono::hours validity (HOURS_IN_DAY * certgenConfig.certConf().getValidity());
+        currentTimestamp = std::chrono::time_point_cast<std::chrono::seconds>(std::chrono::system_clock::now());
+        std::chrono::time_point<std::chrono::system_clock, std::chrono::seconds> endTimestamp = currentTimestamp + validity;
+        config.setValidTo(endTimestamp.time_since_epoch().count());
+
+        config.setCountry(COUNTRY);
+        config.setState(STATE);
+        config.setLocality(LOCALITY);
+        config.setOrganization(ORGANIZATION);
+        config.setOrganizationUnit(ORGANIZATION_UNIT);
+        config.setCommonName(COMMON_NAME);
+        config.setEmail(EMAIL);
+
+        std::list<std::string> ipList = getSystemIPs (AF_INET);
+        ipList.merge (getSystemIPs (AF_INET6));
+        config.setIpList(ipList);
+
+        config.setDnsList(getSystemDomainNames());
+
+        std::string keyType = certgenConfig.keyConf().keyType();
+        std::string certPem;
+        if (keyType == "RSA")
+        {
+            KeyConfigRsaParams *rsaParams = dynamic_cast<KeyConfigRsaParams *>(certgenConfig.keyConf().params().get());
+            Keys keyPair = Keys::generateRSA(rsaParams->rsaLength());
+            certPem = CertificateX509::selfSignSha256(keyPair, config).getPem();
+        }
+        else if (keyType == "EC")
+        {
+            KeyConfigECParams *ecParams = dynamic_cast<KeyConfigECParams *>(certgenConfig.keyConf().params().get());
+            if (ecParams->ecCurveType() == "PRIME256v1")
+            {
+                Keys keyPair = Keys::generateEC (ECKeyType::PRIME256V1);
+                certPem = CertificateX509::selfSignSha256(keyPair, config).getPem();
+            }
+        }
+        else
+        {
+            throw std::runtime_error ("Invalid key type");
+        }
+
+        // TODO: store the certificate in security wallet
```

Comment:
You can create a function static `void store(CertificateX509 & cert, StorageConf & conf)` doing nothing for now

Reply:
Addressed.